src/consensus.rs

```diff
@@ -179,7 +181,24 @@ impl Consensus {
         tls_config: Option<ClientTlsConfig>,
         channel_service: ChannelService,
         runtime: Handle,
+        reinit: bool,
     ) -> anyhow::Result<(Self, Sender<Message>)> {
+        // If we want to re-initialize consensus, we need to prevent other peers
+        // from re-playing consensus WAL operations, as they should already have them applied.
+        // Do ensure that we are forcing compacting WAL on the first re-initialized peer,
+        // which should trigger snapshot transferring instead of replaying WAL.
+        let force_compact_wal = reinit && bootstrap_peer.is_none();
+
+        // On the bootstrap-ed peers during reinit of the consensus
+        // we want to make sure only the bootstrap peer will hold the true state
+        // Therefore we clear the WAL on the bootstrap peer to force it to request a snapshot
+        let clear_wal = reinit && bootstrap_peer.is_some();
+
+        if clear_wal {
+            log::debug!("Clearing WAL on the bootstrap peer to force snapshot transfer");
+            state_ref.clear_wal()?;
+        }
```

Comment:
This seems a bit dangerous in case we _accidentally_ start with `--bootstrap` on the first peer. But, I don't see a good way to safeguard this.

Reply:
intended to be used on copies of the data, not the main database

Reply:
Fair enough. No concerns then.